common/src/main/java/io/netty/util/concurrent/SingleThreadEventExecutor.java

```diff
@@ -948,6 +948,12 @@ public void run() {
                                 }
                             }
 
+                            // Lets remove all FastThreadLocals for the Thread as we are about to terminate and notify
+                            // the future. The user may block on the future and once it unblocks the JVM may terminate
+                            // and start unloading classes.
+                            // See https://github.com/netty/netty/issues/6596.
+                            FastThreadLocal.removeAll();
```

Comment:
Should this happen after `threadLock.release()`/before `STATE_UPDATER.set()`? It seems awaitTermination returns when threadLock is released.

Reply:
good point... let me fix this